### PR TITLE
Implement Bug Bounty page

### PR DESCRIPTION
**Pull request by wackerow**
## Description
- Implements `/bug-bounty` and affiliated components

## Todo
- Fix i18n strings for path breadcrumbs
- Fix image width—should be 24px wide
- Check `<code>` styling compared to Gatsby repo

## Related Issue
https://www.notion.so/efdn/Migrate-React-pages-8acee0dc80d04e55b109e9852c20ef2f?pvs=4

## Preview link
https://deploy-preview-144--ethereum-org-fork.netlify.app/en/bug-bounty

### CODE REVIEW

**Review by fredrik0x — APPROVED** [review]
LGTM. minor nitpick would be that "Bug Bounty Program 🐛" is cut off a bit short at the bottom (the "g" and "y" is a bit short)

File: src/pages/bug-bounty.tsx [hunk 1]
```diff
@@ -0,0 +1,896 @@
...
+      image: useColorModeValue(tekuDark, tekuLight),
+    },
+  ].map((client) => {
+    client.image.width = 60
+    return client
+  })
+
+  const clientsSmall = clients.map((client) => {
+    const thumbnail = client.image as StaticImageData
+    thumbnail.width = 24
```
Comment (minimalsm) [inline src/pages/bug-bounty.tsx]: I'm seeing a minimum width of 60px in prod

Prod:
Screenshot 2023-11-28 at 12 57 07

This PR:
Screenshot 2023-11-28 at 12 57 11

**Review by minimalsm — CHANGES_REQUESTED** [review]

File: src/components/Breadcrumbs/index.tsx [hunk 2]
```diff
@@ -52,33 +52,34 @@ const Breadcrumbs: React.FC<IProps> = ({
...
+            text: <Translation id="page-index-meta-title" />,
           },
         ]
       : []),
     ,
     ...sliced.map((path, idx) => {
       return {
         fullPath: slugChunk.slice(0, idx + 2).join("/") + "/",
-        text: t(path),
+        text: t(path), // TODO: fix i18n strings for path breadcrumbs
```
Comment (pettinarip) [inline src/components/Breadcrumbs/index.tsx]: I think most of the translations for breadcrumbs are in the `common` ns. Maybe we could import the `common` ns and use it here
```ts
const { t } = useTranslation(["common", "page-index"])

...
text: t(path, { ns: "common" })
```

Just an idea, we can tackle this in a separate PR.

File: src/pages/bug-bounty.tsx [hunk 3]
```diff
@@ -0,0 +1,896 @@
...
+export const getStaticProps: GetStaticProps<{}, {}> = async (context) => {
+  const { locale } = context
+
+  // load i18n required namespaces for the given page
+  const requiredNamespaces = getRequiredNamespacesForPath("bug-bounty")
+
+  return {
+    props: await serverSideTranslations(locale!, requiredNamespaces),
+  }
+}
```
Comment (pettinarip) [inline src/pages/bug-bounty.tsx]: We need to pass the `lastDeployDate` prop to display the date correctly.

```suggestion
  const requiredNamespaces = getRequiredNamespacesForPath("bug-bounty")
  const lastDeployDate = getLastDeployDate()

  return {
    props: {
      ...(await serverSideTranslations(locale!, requiredNamespaces)),
      lastDeployDate
    },
  }
}
```

File: src/pages/bug-bounty.tsx [hunk 4]
```diff
@@ -0,0 +1,896 @@
...
+  title: ReactNode
+  link: string
+}
+
+const sortBountyHuntersFn = (a: BountyHuntersArg, b: BountyHuntersArg) => {
+  if (!a.score || !b.score) return 0
+  return b.score - a.score
+}
+
+export const getStaticProps: GetStaticProps<{}, {}> = async (context) => {
```
Comment (pettinarip) [inline src/pages/bug-bounty.tsx]: Lets use the `satisfies` form, like in [the slug file](https://github.com/ethereum/ethereum-org-fork/blob/dev/src/pages/%5B...slug%5D.tsx#L96).
```suggestion
export const getStaticProps = async (context) => {
```

File: src/lib/constants.ts [hunk 5]
```diff
@@ -25,6 +25,7 @@ export const CONTENT_IMAGES_MAX_WIDTH = 800
 export const GITHUB_BASE_API =
   "https://api.github.com/repos/ethereum/ethereum-org-website"
 export const GITHUB_COMMITS_URL = GITHUB_BASE_API + "/commits"
+export const GITHUB_ROOT = `https://github.com/`
```
Comment (pettinarip) [inline src/lib/constants.ts]: nitpick
```suggestion
export const GITHUB_URL = `https://github.com/`
```

File: src/components/Leaderboard.tsx [hunk 6]
```diff
@@ -54,14 +53,12 @@ const Leaderboard: React.FC<IProps> = ({ content, limit = 100 }) => {
...
-          const { name, username, score } = item
-
-          const hasGitHub = username !== ""
-          const avatarImg = hasGitHub
-            ? `${githubUrl}${username}.png?size=40`
-            : "https://github.com/random.png?size=40"
+        .map(({ name, username, score }, idx) => {
+          const hasGitHub = !!username
+          const avatarImg =
+            GITHUB_ROOT + (username || "random") + ".png?size=40"
```
Comment (pettinarip) [inline src/components/Leaderboard.tsx]: 💯

**Review by pettinarip — CHANGES_REQUESTED** [review]

File: src/pages/bug-bounty.tsx (hunk 1 above)
Comment (wackerow) [reply src/pages/bug-bounty.tsx]: Patched!

File: src/components/Breadcrumbs/index.tsx (hunk 2 above)
Comment (wackerow) [reply src/components/Breadcrumbs/index.tsx]: Okay... so turns out we do not have a string for `"bug-bounty": "Bug bounty",` in our `common.json` file yet... This is also a bug in production, but easy fix.. I'll add the string in English and this should get resolved in time with future Crowdin translations.

By adding this it works locally without adding "common" to the namespaces here

**Comment by wackerow** [comment]
Thanks @fredriksvantes! This was a bug in production already, but I added a small style adjustment to fix this, thanks for surfacing!

File: src/components/Breadcrumbs/index.tsx (hunk 2 above)
Comment (wackerow) [reply src/components/Breadcrumbs/index.tsx]: Ooops, I take that back... need to add "common", but only to the `useTranslation` argument array. Will leave it in

**Review by pettinarip — APPROVED** [review]
🎉

**Pull request merged**